### ISSUE TITLE
Data label out of sync between charts with same syncId when using on mobile

### ISSUE THREAD

**Issue by ethanli83** [recharts/recharts#3073]
- [x] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/recharts-issue-template-forked-lndhl6)

### Steps to reproduce
1. create two charts with same syncId
2. open the page on mobile device (can also replicate on dev tool with device mode)
3. tap on one chart to show the data label
4. tap on the other chart

### What is expected?
Data label still show on both charts

### What is actually happening?
Data label show on the first chart but remove from second chart

| Environment | Info |
|---|---|
| Recharts | v2.1.6 |
| React | 18.2.0 |
| System | macOs 12.6.1 |
| Browser | Chrome  |

---
Basically, tapping between charts cause data label displayed incorrectly.

**Comment by ckifer** [comment]
Reproduced on chrome devtools - can't repro on firefox devtools. Can repro on mobile (iOS, Chrome)

![image](https://user-images.githubusercontent.com/25180830/206386705-90f33b5c-2d56-462b-ae1a-6cf2c60f0e4e.png)

**Comment by HHongSeungWoo** [comment]
@ckifer If you think this behavior is a bug, may I handle it?

**Comment by ckifer** [comment]
Assigned!
